Sources/IRGen/Expression/IRSubscriptExpression.swift

```diff
@@ -43,33 +45,34 @@ struct IRSubscriptExpression {
 
     switch subExpr.baseExpression {
     case .identifier:
-      return runtimeFunc(String(baseOffset), indexExpressionCode)
+      return (preamble, runtimeFunc(String(baseOffset), indexExpressionCode.expression))
     case .subscriptExpression(let newBase):
-      return runtimeFunc(nestedStorageOffset(subExpr: newBase,
-                                             baseOffset: baseOffset,
-                                             functionContext: functionContext),
-                         indexExpressionCode)
+      let e = nestedStorageOffset(subExpr: newBase,
+                          baseOffset: baseOffset,
+                          functionContext: functionContext)
+
+      return (preamble + "\n" + e.0, runtimeFunc(e.1, indexExpressionCode.expression))
     default:
       fatalError("Subscript expression has an invalid type")
     }
   }
 
-  func rendered(functionContext: FunctionContext) -> String {
+  func rendered(functionContext: FunctionContext) -> GeneratedCode {
     guard let identifier = baseIdentifier(.subscriptExpression(subscriptExpression)),
       let enclosingType = identifier.enclosingType,
       let baseOffset = functionContext.environment.propertyOffset(for: identifier.name,
                                                                   enclosingType: enclosingType) else {
         fatalError("Arrays and dictionaries cannot be defined as local variables yet.")
     }
 
-    let memLocation: String = nestedStorageOffset(subExpr: subscriptExpression,
+    let (preamble, memLocation): (String, String) = nestedStorageOffset(subExpr: subscriptExpression,
```

Comment:
Pattern matching is nice and all but a structure (either `GeneratedCode` or something else) would be far more explicit and more type safe